### ISSUE TITLE
Raise an error if `pip install` doesn't have `--no-cache-dir` option provided 

### ISSUE THREAD

**Issue by extsoft** [hadolint/hadolint#455]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Let's build an image where the `pip` has the enabled cache

```shell
docker build -t pip-with-cache - <<EOF
FROM python:3.8-alpine
RUN python -m pip install --upgrade pip && python -m pip install flask
EOF
```

Let's build another image where `pip` has the disabled cache

```shell
 docker build -t pip-no-cache - <<EOF
FROM python:3.8-alpine
RUN python -m pip install --no-cache-dir --upgrade pip && python -m pip install --no-cache-dir flask
EOF
```

Let's compare images

```shell
docker image ls --filter reference="pip*"
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
pip-no-cache        latest              64873c9565ed        8 seconds ago       94.7MB
pip-with-cache      latest              205c4c08f43f        51 seconds ago      96.9MB
```

As the output shows, the image with disabled cache has a smaller size. And there is only one package installed. That's why I propose to add a new rule that will require adding `--no-cache-dir` after `pip` install.

### Actual behavior

There is no error.
